File: public/index/cgi-bin/list_files.php
<?php
header('Content-Type: text/html');

// Directory where files are located
$directory = '../uploads'; // Adjust the path to your uploads directory

if (!is_dir($directory)) {
    echo '<option disabled>Uploads directory not found</option>';
    exit;
}

$files = array_diff(scandir($directory), array('.', '..'));

if (empty($files)) {
    echo '<option disabled>No files available</option>';
    exit;
}

foreach ($files as $file) {
    // Ensure that only files are listed (exclude directories)
    if (is_file("$directory/$file")) {
        echo '<option value="' . htmlspecialchars($file) . '">' . htmlspecialchars($file) . '</option>';
    }
}
?>

File: public/index/cgi-bin/list_files.py
#!/usr/bin/env python3
import os

# Directory where the files are stored
upload_dir = './www/main/cgi-bin/uploads/'

# Required HTTP headers
print("Content-Type: text/html\n")

# Generate the HTML for the select options
try:
    files = os.listdir(upload_dir)
    
    if files:
        for file in files:
            filepath = os.path.join(upload_dir, file)
            if os.path.isfile(filepath):
                print(f'<option value="{file}">{file}</option>')
    else:
        print('<option disabled>No files available</option>')

except Exception as e:
    print(f'<option disabled>Error loading files: {str(e)}</option>')